lib/OnyxCache.js

```diff
@@ -179,12 +179,23 @@ class OnyxCache {
      * Remove keys that don't fall into the range of recently used keys
      */
     removeLeastRecentlyUsedKeys() {
-        while (this.recentKeys.size > this.maxRecentKeysSize) {
-            const iterator = this.recentKeys.values();
+        let toRemove = this.recentKeys.size - this.maxRecentKeysSize;
+        if (toRemove <= 0) {
+            return;
+        }
+        const iterator = this.recentKeys.values();
+        const temp = [];
+        while (toRemove > 0) {
             const value = iterator.next().value;
             if (value !== undefined) {
-                this.drop(value);
+                temp.push(value);
             }
+            toRemove--;
+        }
+
+        for (let i = 0; i < temp.length; ++i) {
+            delete this.storageMap[temp[i]];
+            this.recentKeys.delete(temp[i]);
```

Comment:
Out of curiosity, why not use `drop()` here?

Reply:
We don't want to remove key from storageKeys as that's used for getAllKeys. We don't remove the key from the db only from cache.

Reply:
https://github.com/Expensify/react-native-onyx/pull/362